### PR TITLE
Add Take for REEArray

### PR DESCRIPTION
**Pull request by robert3005**

### CODE REVIEW

File: vortex-ree/src/compute.rs [hunk 1]
```diff
@@ -45,3 +52,39 @@ impl ScalarAtFn for REEArray {
...
+            primitive_indices
+                .typed_data::<$P>()
+                .iter()
+                .map(|idx| {
+                    self.find_physical_index(*idx as usize)
+                        .map(|loc| loc as u64)
+                })
+                .collect::<VortexResult<Vec<_>>>()?
+        });
+        take(self.values(), &PrimitiveArray::from(physical_indices))
```
Comment (robert3005) [inline vortex-ree/src/compute.rs:69]: This is the same pattern that happens in Dict array, repeated value take on some dictionary.

**Review by lwwmanning — APPROVED** [review]

File: vortex-ree/src/compute.rs [hunk 2]
```diff
@@ -45,3 +52,39 @@ impl ScalarAtFn for REEArray {
         scalar_at(self.values(), self.find_physical_index(index)?)
     }
 }
+
+impl TakeFn for REEArray {
+    fn take(&self, indices: &dyn Array) -> VortexResult<ArrayRef> {
+        let primitive_indices = flatten_primitive(indices)?;
+        let physical_indices = match_each_integer_ptype!(primitive_indices.ptype(), |$P| {
```
Comment (gatesn) [inline vortex-ree/src/compute.rs:59]: I think this should just be a search_sorted? Possibly this? https://github.com/fulcrum-so/vortex/pull/164

Comment (gatesn) [reply vortex-ree/src/compute.rs:59]: That said, an iterator would allow us to apply the offset inline without allocating a new buffer

Comment (robert3005) [reply vortex-ree/src/compute.rs:59]: This is just search sorted. That’s what find_physical_index does

Comment (gatesn) [reply vortex-ree/src/compute.rs:59]: Yeah, but a lot of function calls, vs search_sorted_many

Comment (robert3005) [reply vortex-ree/src/compute.rs:59]: Sure

**Pull request merged**